internal/database/conf.go

```diff
@@ -117,6 +135,64 @@ func (s *confStore) SiteGetLatest(ctx context.Context) (_ *SiteConfig, err error
 	return tx.getLatest(ctx)
 }
 
+func (s *confStore) ListSiteConfig(ctx context.Context, opt SiteConfigListOptions) ([]*SiteConfig, error) {
+	// Ascending order by default.
+	orderBy := AscendingOrderByDirection
+	if opt.OrderByDirection != "" {
+		orderBy = opt.OrderByDirection
+	}
+
+	q := sqlf.Sprintf(`
+		SELECT
+			 id,
+			 author_user_id,
+			 type,
+			 contents,
+			 created_at,
+			 updated_at
+		 FROM critical_and_site_config
+         WHERE type = 'site'
+		 ORDER BY id `+string(orderBy)+`
+         %s
+        `,
+		opt.LimitOffset.SQL(),
+	)
+
+	rows, err := s.Query(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	history := []*SiteConfig{}
+	for rows.Next() {
+		var c SiteConfig
+		err := rows.Scan(
+			&c.ID,
+			&dbutil.NullInt32{N: &c.AuthorUserID},
+			&c.Type,
+			&c.Contents,
+			&c.CreatedAt,
+			&c.UpdatedAt,
+		)
+		if err != nil {
+			return []*SiteConfig{}, err
+		}
+
+		history = append(history, &c)
+	}
+
+	return history, nil
+}
```

Comment:
I think it's good practice to define a separate `scan*` function _and_ it also allows us to use `basestore` utilities :)

```suggestion
	rows, err := s.Query(ctx, q)
	return scanSiteConfigs(rows, err)
}

var scanSiteConfigs = basestore.NewSliceScanner(scanSiteConfig)

func scanSiteConfig(s dbutil.Scanner) (*SiteConfig, error) {
	var c SiteConfig
	err := s.Scan(
		&c.ID,
		&dbutil.NullInt32{N: &c.AuthorUserID},
		// &c.Type, -- this doesn't exist, right?
		&c.Contents,
		&c.CreatedAt,
		&c.UpdatedAt,
	)
	if err != nil {
		return nil, err
	}
	return &c, nil
}
```